static/app/utils/useSyncedLocalStorageState.tsx

```diff
@@ -0,0 +1,49 @@
+import {useCallback, useEffect} from 'react';
+
+import {useLocalStorageState} from 'sentry/utils/useLocalStorageState';
+
+type SyncedLocalStorageEvent = CustomEvent<{key: string; value: any}>;
+
+const SET_VALUE_EVENT_NAME = 'synced-local-storage';
```

Comment:
maybe `const SYNCED_STORAGE_EVENT` or similar since it's not strictly related to setting a value?